### PR TITLE
Updated translation of Chapter 6 in Romanian 🇷🇴

### PR DESCRIPTION
**Pull request by alcyone33**
- Added missing diacritics
- Updated the code sections
- Improved the wording in several places

### CODE REVIEW

**Comment by alexandruradovici** [comment]
~Duplicat https://github.com/UPB-CS-OpenSourceUpstream/tour_of_rust/pull/7, te rog sa te uiti ce exista deja.~

never mind :)

**Comment by alexandruradovici** [comment]
Il poti trimite catre repository-ul original si pune un link aici te rog.

**Review by alexandruradovici — APPROVED** [review]

**Comment by alcyone33** [comment]
Pull request repo original:

https://github.com/richardanaya/tour_of_rust/pull/491

File: lessons/ro/chapter_6.yaml [hunk 1]
```diff
@@ -129,25 +148,25 @@
...
-    * `trim` pentru eliminarea spatiilor.
+    * `trim` pentru eliminarea spațiilor.
 
 
     Atunci când un șir de caractere este eliminat (drop), este eliminată și memoria heap a acestuia.
 
 
-    Șirile de caractere au un operator `+` care extinde șirul cu un `&str`. (Șirul din stânga este `consumat`, iar buffer-ul lui
-    este refolosit pentru a stoca valoarea finală. Șirul din dreapta este doar împrumutat pentru a efectua concatenarea.)
+    Șirile de caractere au un operator `+` care extinde șirul cu un `&str` (șirul din stânga este `consumat`, iar buffer-ul lui
```
Comment (octavianarmasu) [inline lessons/ro/chapter_6.yaml:157]:
```suggestion
    Șirurile de caractere au un operator `+` care extinde șirul cu un `&str` (șirul din stânga este `consumat`, iar buffer-ul lui
```
Typo la scrierea cuvantului cuvantului "sirurile"

**Pull request closed without merging**